projects/Mallard/src/hooks/use-edition-provider.tsx

```diff
@@ -0,0 +1,260 @@
+import React, {
+    createContext,
+    useContext,
+    useEffect,
+    useState,
+    Dispatch,
+} from 'react'
+import { RegionalEdition, SpecialEdition } from 'src/common'
+import { eventEmitter } from 'src/helpers/event-emitter'
+import { defaultSettings } from 'src/helpers/settings/defaults'
+import {
+    defaultEditionCache,
+    selectedEditionCache,
+    editionsListCache,
+} from 'src/helpers/storage'
+import { errorService } from 'src/services/errors'
+import { defaultRegionalEditions } from '../../../Apps/common/src/editions-defaults'
+import NetInfo from '@react-native-community/netinfo'
+import { AppState, AppStateStatus } from 'react-native'
+import { remoteConfigService } from 'src/services/remote-config'
+import { locale } from 'src/helpers/locale'
+
+interface EditionsEndpoint {
+    regionalEditions: RegionalEdition[]
+    specialEditions: SpecialEdition[]
+}
+
+interface EditionState {
+    editionsList: EditionsEndpoint
+    selectedEdition: RegionalEdition | SpecialEdition
+    defaultEdition: RegionalEdition
+    storeSelectedEdition: (
+        chosenEdition: RegionalEdition | SpecialEdition,
+        type: 'RegionalEdition' | 'SpecialEdition' | 'TrainingEdition',
+    ) => void
+}
+
+export interface StoreSelectedEditionFunc {
+    (
+        chosenEdition: RegionalEdition | SpecialEdition,
+        type: 'RegionalEdition' | 'SpecialEdition' | 'TrainingEdition',
+    ): void
+}
+
+export const DEFAULT_EDITIONS_LIST = {
+    regionalEditions: defaultRegionalEditions,
+    specialEditions: [],
+}
+
+export const BASE_EDITION = defaultRegionalEditions[0]
```

Comment:
Should we default it to UK even when user device locale is USA or AUS?

Reply:
Default editions are what we use on app start up to determine which edition to use. The initial state is set to UK. We then check immediately if there is anything in async storage for a default edition. If not, we check to see if the remote config locale switch is on. If it is, we then determine based on locale a default edition and then set that as selected

That is all done here: https://github.com/guardian/editions/pull/1326/files#diff-2f2c9fc8980b6bc14e3b5220c9156afaR140

So yes, we default to US and AUS if their locale is set to AUS. Unless they then choose a different regional edition, then that becomes the default.

Defaults can only be Regional Editions